### PR TITLE
Floating tile mix

### PR DESCRIPTION
**Pull request by dmh43**
Implements support for float windows in tile groups!
I haven't tested with multiple heads yet.
I also have plans to reorganize some of code between `tile-group.lisp` `floating-group.lisp` and `group.lisp` so that I don't have to reach into `stumpwm.floating-group::float-window`.

### CODE REVIEW

**Comment by dbjergaard** [comment]
This is awesome! Give me a bit to review the code and I'll merge it ASAP.

**Comment by dmh43** [comment]
@jorams, I agree for the most part, but I believe that a purely floating group is still useful occasionally, so I'd like to keep that around. I'll spend some time tomorrow cleaning up the organization.

File: floating-group.lisp [hunk 1]
```diff
@@ -135,12 +135,17 @@
   (float-window-align window)
   (focus-window window))
 
+(defmethod group-add-window (group (window float-window) &key &allow-other-keys)
```
Comment (dmh43) [inline floating-group.lisp]: I'd like to refactor some of the code to remove the use of multimethods where it doesn't seem entirely appropriate. What do you think? See: https://github.com/stumpwm/stumpwm/issues/300

**Comment by dmh43** [comment]
@jorams I've cleaned up the code a bit. I'm not entirely satisfied with the organization so I've made https://github.com/stumpwm/stumpwm/issues/301 to address the code organization in the project in general. Are you alright addressing the code organization concerns in a separate PR?

**Comment by jorams** [comment]
I responded to the issues you created. In summary, I'm definitely alright adressing the code organization in a separate PR, but removal of the generic functions less so. At least `group-add-window` should definitely stay a generic function.

File: floating-group.lisp (hunk 1 above)
Comment (dmh43) [inline floating-group.lisp]: ^ this has been resolved. Organization will be addressed in a separate PR.

**Comment by dbjergaard** [comment]
is this good to go?

**Comment by dmh43** [comment]
@dbjergaard @jorams @PuercoPop I've been using this daily and haven't run into any issues so far. If there aren't any more comments on the code then I think it's good to go.

Also, I haven't tested this with multiple heads, but just by looking at the code I dont expect any issues.

**Comment by dbjergaard** [comment]
@jorams please merge this when you are happy.

**Comment by jorams** [comment]
Alright, merging. The one thing that still stands out is that the floating group should be merged back into the `stumpwm` package, but that can wait.

File: group.lisp [hunk 2]
```diff
@@ -101,6 +101,14 @@ needs to redraw anything on it, this is where it should do it."))
   (:documentation "When a head or its usable area is resized, this is
 called. When the modeline size changes, this is called."))
 
+(defmethod group-current-window (group)
+  (screen-focus (group-screen group)))
```
Comment (jorams) [inline group.lisp]: This is causing an issue. The "current window" in a group can now be a window that's not actually in that group, which goes against expectations in other parts of the codebase. (Currently I'm looking at `group-forward-with-window`)

Comment (dmh43) [inline group.lisp]: Ah, thanks for catching this! I think the best way to address this would be to keep track of the current window on the group object itself. What do you think?

Comment (jorams) [inline group.lisp]: I think so too, and that's why I opened #324 :)
